src/error.rs

```diff
@@ -26,6 +26,21 @@ pub enum RequestError {
     },
 }
 
+impl RequestError {
+    pub fn as_response(&self) -> Response<Body> {
+        let status_code = match self {
+            RequestError::AcquiringTicket { .. } | RequestError::InvalidURI { .. } => 500,
+            RequestError::InvalidMethod { .. } | RequestError::InvalidPath { .. } => 501,
+            RequestError::RequestIssue { .. } => 502,
+        };
+
+        Response::builder()
+            .status(status_code)
+            .body(Body::empty())
+            .unwrap()
```

Comment:
Could we include a body that contains the error as well, maybe with a `http-proxy` prefix

Reply:
To quote the RFC we should have it:
> 10.5 Server Error 5xx
>
 >  Response status codes beginning with the digit "5" indicate cases in
   which the server is aware that it has erred or is incapable of
   performing the request. Except when responding to a HEAD request, the
   server SHOULD include an entity containing an explanation of the
   error situation, and whether it is a temporary or permanent
   condition. User agents SHOULD display any included entity to the
   user. These response codes are applicable to any request method.

Reply:
done